### ISSUE TITLE
PVM does not work with SGE under SUSE Linux 10.0

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#216]
**Reported by bli on 25 Jul 2006 05:35 UTC**
With branch-5-0 r5215, PVM does not work with SGE under SUSE Linux 10.0.

pvmtest.err looks like this:

```
[pid6595](pvmd) 07/24 22:29:04 usage: pvmd3 [[-nhostname](-ddebugmask]) [pid6595](hostfile]
[pvmd) 07/24 22:29:04 pvmbailout(0)
libpvm [/tmp/4.1.all.q/pvmd.1000: No such file or directory
libpvm [pid6559](pid6559]) /tmp/4.1.all.q/pvmd.1000: No such file or directory
libpvm [pid6559] /tmp/4.1.all.q/pvmd.1000: No such file or directory
```

and pvmtest.out is:

```
-catch_rsh /opt/sge/default/spool/nano01/active_jobs/4.1/pe_hostfile oscar01.ocg.org /opt/pvm3
/opt/sge/bin/lx26-x86/qrsh -V -inherit oscar01.ocg.org env PVM_TMP=$TMPDIR /opt/pvm3/lib/pvmd /tmp/4.1.all.q/hostfile -f
startpvm.sh: startup failed - invoking cleanup script
-catch_rsh /opt/sge/default/spool/nano01/active_jobs/4.1/pe_hostfile oscar01.ocg.org
-catch_rsh /opt/sge/default/spool/nano01/active_jobs/4.1/pe_hostfile oscar01.ocg.org
```

Output of hostname of one node:

```
# hostname
nano01
```

This is probably caused by the inconsistency between "oscar01" and "oscar01.ocg.org".  May need to be fixed in the PVM-SGE integration scripts level...

**Comment by dikim33** [comment]
**Comment by bli on 25 Jul 2006 06:33 UTC**
The code in question is in $SGE_ROOT/pe/pvm/rsh:

```
if [ "$rhost" != "`hostname`" ] ; then
```

The call to `hostname` needs to be changed to `hostname --long` otherwise you'll be comparing shortname with FQDN under SUSE Linux.

The SGE RPMs need to be rebuilt, unfortunately for this fix...

**Comment by dikim33** [comment]
**Comment by bli on 9 Aug 2006 08:40 UTC**
I have a working RPM (6.0u8-5) that works with SUSE - sent to !DongInn for testing, if it pass then I'll check it into both trunk & branch-5-0.

**Comment by dikim33** [comment]
**Comment by bli on 29 Aug 2006 07:47 UTC**
(In [5323]) This commit is a merge from trunk -> branch-5-0:
   svn merge -r5321:5322 https://svn.oscar.openclustergroup.org/svn/oscar/trunk .

Closes #217 for milestone OSCAR 5.0.

Original commit message:
Fix for #217.  SGE-PVM integration now works under SUSE Linux 10.0.  This revision for SGE RPM
also builds libdrmaa.so.  Needs rebuilding on _all_ supported distroes/archs.
